test/onnx/parse/pad_undef_const_val_test.cpp

```diff
@@ -0,0 +1,38 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2015-2024 Advanced Micro Devices, Inc. All rights reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include <onnx_test.hpp>
+
+TEST_CASE(pad_undef_const_val_test)
+{
+    migraphx::program p;
+    auto* mm = p.get_main_module();
+    auto l0  = mm->add_parameter("0", migraphx::shape{migraphx::shape::float_type, {2, 2}});
+    mm->add_literal({migraphx::shape{migraphx::shape::int32_type, {4}}, {1, 1, 1, 1}});
+    mm->add_instruction(migraphx::make_op("undefined"));
```

Comment:
This comes from pytorch: https://github.com/pytorch/pytorch/issues/84979
By default `value` is not set (`None`) for Pad. During torch export the `None` becamues an empty input.
That empty input becomes an `undefined` during parsing here: 
https://github.com/ROCm/AMDMIGraphX/blob/4b9d647041636a1ec54265b8a71686a0d0aac25f/src/onnx/onnx_parser.cpp#L400-L403

Reply:
May I add a comment abot this issue with torch export?

Reply:
For example : 
https://github.com/ROCm/AMDMIGraphX/blob/4b9d647041636a1ec54265b8a71686a0d0aac25f/test/dead_code_elimination_test.cpp#L137

Reply:
AFAIK DCE runs after the parsing, and this `undefined` is during parse.

Reply:
@pfultz2  do you know why "undefind" remains here despite running DCE ?

Reply:
> AFAIK DCE runs after the parsing, and this undefined is during parse. 

Yes after the parsing `optimize_onnx` would run DCE
https://github.com/ROCm/AMDMIGraphX/blob/4b9d647041636a1ec54265b8a71686a0d0aac25f/test/onnx/include/onnx_test.hpp#L47

Reply:
DCE is not running because `run_passes` was not set for `optimize_onnx` in the test.
https://github.com/ROCm/AMDMIGraphX/blob/4b9d647041636a1ec54265b8a71686a0d0aac25f/test/onnx/include/onnx_test.hpp#L41-L49

Reply:
Oh. Right. I don't know what i was thinking.